src/jit/optimizer.cpp

```diff
@@ -7837,7 +7837,9 @@ GenTree *           Compiler::optIsBoolCond(GenTree *   condBranch,
     if  (opr2->gtOper != GT_CNS_INT)
         return  NULL;
 
```

Comment:
I'd want to avoid intermixing correctness fixes with random CC improvements. For this function I would have to make dozens of edits to get it up to the CC standard so that it would be consistent with itself. We should fix all of this CC stuff in a separate wide-spread change.